hadoop-hdds/common/src/main/java/org/apache/hadoop/util/MetricUtil.java

```diff
@@ -17,17 +17,32 @@
  */
 package org.apache.hadoop.util;
 
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
 import org.apache.hadoop.metrics2.lib.MutableRate;
 import org.apache.ratis.util.function.CheckedRunnable;
 import org.apache.ratis.util.function.CheckedSupplier;
 
 import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
 import java.util.function.Consumer;
 
 /**
  * Encloses helpers to deal with metrics.
  */
 public final class MetricUtil {
+
+  private static final ExecutorService EXECUTOR =
+      new ThreadPoolExecutor(1, 10, 100, TimeUnit.MILLISECONDS,
+          new LinkedBlockingQueue<>(),
```

Comment:
provide the queue limit for this to avoid OOM

Reply:
Is Integer.MAX_VALUE too much for the case?

Reply:
I may store upto 2G of objects, and each object may cost 64+bytes (long, function reference) at least, can consume atleast 128GB of heap usages.

Reply:
If metrics getting produced at very high rate then being consumed, then this can cause OOM as above case. So better limit the queue size.